src/Kernel/client/ExecutionPathVisualizer/metadata.ts

```diff
@@ -13,6 +13,9 @@ export interface Metadata {
     controlsY: number[];
     /** Array of y coords of target registers. */
     targetsY: number[];
+    /** y coords of target registers separated by classical registers.
+     *  This is used for rendering multi-qubit unitary gates. */
+    groupedTargetsY?: number[][];
```

Comment:
Is `targetsY` still needed on this interface? It seems like `groupedTargetsY` should contain the same information (plus additional information about the grouping), or is there a case I'm missing?

Are you doing it this way to avoid breaking the existing consumer of this interface?

Reply:
Yea, that's a good question which I should have explained in the description. `groupedTargetsY` is only used for gates of type `GateType.Unitary` as they are the only gates that span multiple registers as big unitary box. Other gate types don't need to be split, so they use `targetsY`. `groupedTargetsY` is only computed for `GateType.Unitary` currently. There might be a way to merge them both 🤔

Reply:
I see. I am wondering if perhaps `targetsY` could be used for both scenarios. It could be a `number[]`, in which case you assume the simple scenario, or `number[][]`, in which case you treat it as `groupedTargetsY`?

This isn't really such a big deal, just nice to avoid requiring redundant information from callers if possible.

Reply:
That makes sense, I'll give it a try :)